### PR TITLE
Square bracket list initialization syntax

### PR DESCRIPTION
**Pull request by pboyer**
### Purpose

This PR adds new syntax to initialize lists with square brackets - a la JS or Python. 

```
a = []; // empty list
b = [1,2,3]; // list of 3 numbers
```

This PR doesn't deprecate the old syntax for initializing lists. So, you can still do this:

```
a = {}; // empty list
b = {1,2,3}; // list of 3 numbers
```

A future PR will deprecate this syntax while migrating old code blocks by code rewriting to the new syntax.

screen shot 2018-01-04 at 11 36 59 am

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@aparajit-pratap 

### FYIs

@kronz @Racel

### CODE REVIEW

**Comment by pboyer** [comment]
@aparajit-pratap How would you suggest we test this? I could change existing test cases in the DS test suite to use the new syntax, but I don't plan on supporting both syntaxes for long...

File: src/Engine/ProtoCore/ExtendedLibraries/FunctionObject.ds [hunk 1]
```diff
@@ -32,42 +32,42 @@ def __CreateFunctionObject(
 	fo = Function();
 	imp = { "__funcobj" : fobj };
 	fo.imp = imp;
-    return = fo;
+    return fo;
```
Comment (pboyer) [inline src/Engine/ProtoCore/ExtendedLibraries/FunctionObject.ds]: I went through this file and updated to newer syntax.

**Comment by aparajit-pratap** [comment]
@pboyer if we aren't supporting both syntaxes for long, we'll need to change existing test cases from `{}` to `[]` unless we can also auto-migrate those tests.

File: src/Engine/ProtoCore/Parser/Parser.cs [hunk 2]
```diff
@@ -176,6 +176,66 @@ private bool IsIdentList()
         return false;
     }
 
+    private bool IsListExpression()
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/Parser/Parser.cs]: I don't see this method in the ATG?

**Comment by aparajit-pratap** [comment]
@pboyer other than the removal of the unused method in `parser.cs`, I think this PR is good. Thanks.

File: src/Engine/ProtoCore/Parser/Parser.cs (hunk 2 above)
Comment (pboyer) [reply src/Engine/ProtoCore/Parser/Parser.cs]: Good catch, @aparajit-pratap! I deleted that at the last minute from the atg as it was unused. I'll regen.

**Comment by aparajit-pratap** [comment]
@pboyer curious how you did the untabification if not manually.

**Comment by pboyer** [comment]
In MSVS : Edit > Advanced > Untabify selection

**Pull request merged**